src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -95,13 +109,21 @@ public void Stop()
             this.logger.LogTrace("(-)");
         }
 
-        /// <inheritdoc />
-        public virtual void ProcessBlock(Block block)
+        /// <summary>Called when a <see cref="Block"/> is added to the <see cref="blocksQueue"/>.
+        /// Depending on the <see cref="WalletTip"/> and incoming block height, this method will decide whether the <see cref="Block"/> will be processed by the <see cref="WalletManager"/>.
+        /// </summary>
+        /// <param name="block">Block to be processed.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        private async Task OnProcessBlockAsync(Block block, CancellationToken cancellationToken)
         {
             Guard.NotNull(block, nameof(block));
             this.logger.LogTrace("({0}:'{1}')", nameof(block), block.GetHash());
 
+            if (block.BlockSize != null)
```

Comment:
This should not be needed because there should be a guarantee that the block size is calculated properly every time. Only in case of some incomplete test setup, this can be null

Reply:
ok